README.md

```diff
@@ -1,11 +1,63 @@
-# QueryForge
-We intend to develop a web application that conducts an exhaustive search of relevant professors, their current research endeavors, contact information, ongoing projects, and their openness to hiring undergraduate, graduate, and post-doctoral student researchers, among other pertinent details aligned with the research interests of students. 
+## QueryForge
+This web application aims to streamline the process of discovering research opportunities within a university setting. By aggregating data on professors, research projects, and student positions from multiple sources, this tool allows students to search for research projects relevant to their interests. 
 
-Team Memebers
+As students, we understand the challenge of finding research opportunities due to scattered information. This tool provides a centralized database with enhanced search and filtering functionalities, tailored to support students in the ECE department, with plans to expand to additional departments.
+
+## Team Memebers
 
 * Stuti Rastogi
 * Disha Maheshwari
 
-To run
+
+## Features
+- **Research Opportunity Aggregation**: Gathers data on professors, research projects, and student openings across the ECE department.
+- **Resume Parser**: Enables users to upload their resumes, extract key skills and qualifications, and match them with relevant research projects.
+- **Advanced Search and Filtering**: Allows users to search and filter opportunities based on research interests, professor details, and project type.
+
+
+## Architecture
+The application follows a multi-tier architecture, consisting of:
+1. **Frontend**: User interface for searching and filtering research opportunities.
+2. **Backend**: RESTful API for data handling, including resume parsing and data aggregation.
+3. **Database**: Stores all research data, professor details, and user information.
+4. **Web Scraping Module**: Pulls data from university websites and professor pages using `BeautifulSoup` and `Requests`.
+5. **Resume Parser Module**: Extracts and analyzes text from user resumes to match with relevant projects.
+
+### Prerequisites
+- **Python 3.8+**
+- **Flask** (for backend)
+- **BeautifulSoup** and **Requests** (for web scraping)
+- **NLTK** (for resume parsing)
+- **PyPDF2** and **pdfplumber** (for PDF handling)
+  
+
+## Usage
+1. **Search for Opportunities**: Use the search bar and filters on the homepage to find research opportunities by professors or project areas.
+2. **Upload Resume**: Navigate to the Resume section to upload your resume, analyze key terms, and find matching projects.
+3. **Refine Results**: Use additional filters to narrow down results based on specific departments, VIP/EPICS project teams, or available positions.
+
+## Technical Stack
+- **Frontend**: HTML, CSS, JavaScript
+- **Backend**: Flask, REST API
+- **Database**: SQLite/PostgreSQL
+- **Web Scraping**: BeautifulSoup, Requests
+- **Resume Parsing**: PyPDF2, pdfplumber, NLTK
```

Comment:
Add details about the RAG model. RAG architecture details are missing.

Reply:
Done!